doc/contributing/maintaining-shared-library-support.md

```diff
@@ -0,0 +1,119 @@
+# Maintaining shared library support
+
+Node.js unofficially supports a build option where Node.js is built as
+a shared library (node.dll/libnode.so). The shared library provides
+an easy way to embed Node.js into other applications and to
+have multiple applications use a single copy of Node.js instead
+of having to bundle in the full Node.js footprint into each application.
+For workloads that require multiple node instances this can result
+in a significant footprint savings.
+
+This doc provides an outline of the approach and things to look
+out for when maintaining the shared library support.
+
+Currently shared library support has only be tested on:
+
+* Linux
+* macOS
+* Windows
+* AIX
+
+## Building with shared library option
+
+Node.js is built with the shared library option by adding `--shared`
+to the configure step before building on non-windows platforms
+and adding `dll` when building on windows.
```

Comment:
This paragraph is particularly convoluted. I've tried a suggestion to restructure it a few times, but I'm not familiar enough with the subject to be able to confidently do that. I'd recommend trying to rewrite it for clarity.